_docs-sources/pipelines/overview/configurations-as-code.md

```diff
@@ -0,0 +1,390 @@
+
+At a high level, Pipelines will read these configurations by parsing all files that end with `.hcl` within a directory named `.gruntwork` or a single file named `gruntwork.hcl`. In typical usage, the configurations that are global to the IAC in a repository will be defined within a `.gruntwork` directory at the root of the repository, and configurations that are specific to a particular `terragrunt.hcl` file (a unit) located in the same directory as the `terragrunt.hcl` file.
+
+## Minimum Required Configuration
+
+The minimum configurations required for Pipelines to operate correctly will vary depending on context. For the most common usage, Pipelines will need to be able to determine how to authenticate with a cloud provider in order to run Terragrunt commands. If it is not able to determine how to authenticate within a context where it needs to be able to do so, Pipelines will throw an error.
+
+The following is an example of a minimal configuration that would allow Pipelines to determine how to authenticate with AWS for exactly one unit of IAC:
+
+```hcl
+# gruntwork.hcl
+unit {
+  authentication {
+    aws_oidc {
+      account_id = "an-aws-account-id"
+      authentication_profile = {
+        plan_iam_role = "role-to-assume-for-plans"
+        apply_iam_role = "role-to-assume-for-applies"
+      }
+    }
+  }
+}
+```
+
+Placing this configuration in a `gruntwork.hcl` file in the same directory as a `terragrunt.hcl` file will cause Pipelines to assume the `role-to-assume-for-plans` role in the AWS account with ID `an-aws-account-id` when running Terragrunt plan commands, using [OIDC](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) to authenticate to AWS and assume that role.
+
+In most circumstances, the same role would be assumed by multiple units of configuration within a repository (e.g. all units within a given directory configure resources for the same AWS account). Hence, it would be more appropriate to define a `.gruntwork` directory at the root of the repository with configuration files ending in `.hcl` that are applicable to all units within a given directory.
+
+e.g.
+
+```hcl
+# .gruntwork/environments.hcl
+environment "an_environment" {
+  filter {
+    paths = ["an-environment/*"]
+  }
+
+  authentication {
+    aws_oidc {
+      account_id = "an-aws-account-id"
+      authentication_profile = {
+        plan_iam_role = "role-to-assume-for-plans"
+        apply_iam_role = "role-to-assume-for-applies"
+      }
+    }
+  }
+}
+```
+
+In this example, all the units located within the `an-environment` directory sibling to the `.gruntwork` directory will assume the `role-to-assume-for-plans` role in the AWS account with ID `an-aws-account-id` when running Terragrunt plan commands by Pipelines.
```

Comment:
I think it might actually be better to avoid that, because `dev` is more general than what our environments actually relate to. If you have two products, each in their own AWS accounts, and they each have their own dev, stage and prod accounts as part of their SDLC lifecycle, you wouldn't want to think it's appropriate to have a `dev` environment.

You would actually want `product1_dev` and `product2_dev` environments.